Modified conversion of elements related to MPolyQuo

raulepure opened the pull request:
In trying to fix some segfault errors, I removed "_badpolymap"
and adapted the conversion of elements related to MPolyQuo.
I had to change the code in the normalization a little. 
@tthsqe12:  I hope it is still working as it should.

Commit:
Modified conversion of elements related to MPolyQuo

In trying to fix some segfault errors, I removed "_badpolymap"
and adapted the conversion of elements related to MPolyQuo.

tthsqe12 commented:
I am glad you removed badpolymap and am interested in what you replaced it with. Is it possible to add tests for cases that were crashing?

raulepure commented:
I replaced _badpolymap just by using the conversions which are already there, which map MPolyElem to Singular.spoly and back. 
I just added some checks regarding number of variables and parents, where needed, to avoid too many conversions which cause trouble. Still not the best thing, which can be done, but it removed some errors which one can cause involuntarily.

Regarding the tests, I excluded the bad examples, since I forbid something as follows:
```
julia> r, (x, y, z, t) = PolynomialRing(QQ, ["x", "y", "z", "t"])
(Multivariate Polynomial Ring in x, y, z, t over Rational Field, fmpq_mpoly[x, y, z, t])

julia> s, (a,b,c) = PolynomialRing(QQ, ["a", "b", "c"])
(Multivariate Polynomial Ring in a, b, c over Rational Field, fmpq_mpoly[a, b, c])

julia> S = quo(s, ideal(s, [c-b^3]))[1]
SxQuotient of Multivariate Polynomial Ring in a, b, c over Rational Field by ideal generated by: -b^3 + c

julia> Sx = Oscar.singular_ring(S)
Singular Polynomial Quotient Ring (QQ),(a,b,c),(dp(3),C)

julia> r(Sx(S(a).f))
x*t^139669453035008
```
Once I add a check that the numbers of variables are identical, this does not occur anymore.
I'm not sure how to test it. I would gladly like to add a test.

tthsqe12 commented:
Ok. Sanitizing something like `r(Sx(S(a).f))` is not a priority since using the private `.f` member means you are a developer who knows what they are doing.
